torch/distributed/_sharded_tensor/ops/linear.py

```diff
@@ -0,0 +1,129 @@
+import torch
+import torch.distributed as dist
+from torch.distributed._sharding_spec import ChunkShardingSpec
+from torch.distributed._sharding_spec._internals import (
+    get_split_size,
+    get_chunked_dim_size,
+)
+from typing import List
+
+
+def sharded_linear(types, args, kwargs, pg):
+    from torch.distributed._sharded_tensor import ShardedTensor
+
+    input = args[0]
+    weight = args[1]
+    bias = kwargs["bias"]
+
+    # Validate types
+    if not isinstance(input, torch.Tensor) or not isinstance(bias, torch.Tensor):
+        raise TypeError("input and bias need to be torch.Tensor")
+    if not isinstance(weight, ShardedTensor):
+        raise TypeError("weight needs to be ShardedTensor")
+
+    if input.size()[1] != weight.size()[1]:
+        raise ValueError(
+            f'Input dim: {input.size()[1]} does not match '
+            f'appropriate weight dim: {weight.size()[1]}')
+    if not isinstance(weight._sharding_spec, ChunkShardingSpec):
+        raise ValueError("Only ChunkShardingSpec supported for ShardedTensor ops!")
+    if len(weight.local_shards()) != 1:
+        raise ValueError("Only one local shard supported!")
```

Comment:
why we are only supporting the case that each rank have one local rank? is it because we need to preserve additional information when doing all_gather and it seems in-feasible to broadcast that?